### PR TITLE
Update SUNDIALS to 7.1.1

### PR DESCRIPTION
**Pull request by jrood-nrel**

### CODE REVIEW

**Comment by jrood-nrel** [comment]
https://github.com/AMReX-Combustion/PelePhysics/issues/530

**Comment by jrood-nrel** [comment]
https://github.com/AMReX-Combustion/PeleC/pull/854 is testing this for PeleC.

**Review by baperry2 — APPROVED** [review]
ReactEval gives identical results before and after this PR (both with and without use_erksep).

The PeleC test that is failing should pass if you pull in the latest PelePhysics changes where I fixed that nullptr issue.

File: Source/Reactions/ReactorArkode.cpp [hunk 1]
```diff
@@ -199,31 +199,31 @@ ReactorArkode::react(
   if (use_erkstep == 0) {
```
Comment (marchdf) [inline Source/Reactions/ReactorArkode.cpp:199]: I think we can consolidate this now. Just the `arkode_mem = ` line in these 2 branches are different.

File: Source/Reactions/ReactorUtils.H [hunk 2]
```diff
@@ -147,9 +147,9 @@ set_sundials_solver_tols(
   if (solvername == "cvode") {
     flag = CVodeSVtolerances(sundials_mem, relTol, atol);
   } else if (solvername == "arkstep") {
```
Comment (marchdf) [inline Source/Reactions/ReactorUtils.H]: same here, we don't need the 2 branches, the code is the same.

File: Source/Reactions/ReactorArkode.cpp [hunk 3]
```diff
@@ -199,31 +199,31 @@ ReactorArkode::react(
...
       ERKStepCreate(cF_RHS, time, y, *amrex::sundials::The_Sundials_Context());
-    ERKStepSetUserData(arkode_mem, static_cast<void*>(user_data));
+    ARKodeSetUserData(arkode_mem, static_cast<void*>(user_data));
     utils::set_sundials_solver_tols<Ordering>(
       *amrex::sundials::The_Sundials_Context(), arkode_mem, user_data->ncells,
       relTol, absTol, m_typ_vals, "erkstep", verbose);
     ERKStepSetTableNum(arkode_mem, static_cast<ARKODE_ERKTableID>(rk_method));
-    int flag = ERKStepSetAdaptController(arkode_mem, sun_controller);
+    int flag = ARKodeSetAdaptController(arkode_mem, sun_controller);
     utils::check_flag(&flag, "ERKStepSetAdaptController", 1);
```
Comment (marchdf) [inline Source/Reactions/ReactorArkode.cpp]:
```suggestion
    utils::check_flag(&flag, "ARKodeSetAdaptController", 1);
```

File: Source/Reactions/ReactorArkode.cpp [hunk 4]
```diff
@@ -199,31 +199,31 @@ ReactorArkode::react(
...
-    ARKStepSetUserData(arkode_mem, static_cast<void*>(user_data));
+    ARKodeSetUserData(arkode_mem, static_cast<void*>(user_data));
     utils::set_sundials_solver_tols<Ordering>(
       *amrex::sundials::The_Sundials_Context(), arkode_mem, user_data->ncells,
       relTol, absTol, m_typ_vals, "arkstep", verbose);
     ARKStepSetTableNum(
       arkode_mem, ARKODE_DIRK_NONE, static_cast<ARKODE_ERKTableID>(rk_method));
-    int flag = ARKStepSetAdaptController(arkode_mem, sun_controller);
+    int flag = ARKodeSetAdaptController(arkode_mem, sun_controller);
     utils::check_flag(&flag, "ARKStepSetAdaptController", 1);
```
Comment (marchdf) [inline Source/Reactions/ReactorArkode.cpp]:
```suggestion
    utils::check_flag(&flag, "ARKodeSetAdaptController", 1);
```

File: Source/Reactions/ReactorArkode.cpp (hunk 1 above)
Comment (jrood-nrel) [reply Source/Reactions/ReactorArkode.cpp:199]: These look different enough to me to just keep them. I think I could pull like 1 line out but then have to think of the BL_PROFILE stuff so I'm going to leave it.

**Review by marchdf — APPROVED** [review]

**Pull request merged**